bmap/src/lib.rs

```diff
@@ -79,3 +78,17 @@ where
 
     Ok(())
 }
+
+pub fn copy_nobmap<I, O>(input: &mut I, output: &mut O) -> Result<(), CopyError>
+where
+    I: Read + SeekForward,
+    O: Write + SeekForward,
+{
+    let mut v = Vec::new();
+    let r = input.read_to_end(&mut v).map_err(CopyError::ReadError)?;
```

Comment:
you're reading all of the file into memory and then writing out again, which is super inefficient

Usually (like in the case of the original `bmap::copy`) we read small chunks into memory at a time, then write those out.

It'd be interesting to try to use https://doc.rust-lang.org/std/io/fn.copy.html here as that handles the buffering for us?

Reply:
I just updated the code to use `st::io::copy` and without the `Seekforward` requirements for I and O. I've tested and, as before, only works with some files. I don't know why yet.

Reply:
Another possibility is that the difference is originated in the output file not being truncated since we don't have the output size at that moment.